pkg/kubectl/cmd/annotate.go

```diff
@@ -193,7 +193,11 @@ func (o AnnotateOptions) RunAnnotate() error {
 			return err
 		}
 
-		name, namespace, obj := info.Name, info.Namespace, info.Object
+		obj, err := resource.AsVersionedObject([]*resource.Info{info}, false, info.Mapping.GroupVersionKind.Version)
```

Comment:
this is sort of going around our elbow to reach our nose... we're using the GroupVersionKind from the mapping, which is not what we've negotiated with the server. If we're just going to use the info the mapping, we should just use `info.VersionedObject.(runtime.Object)` instead